src/Cake.Common/Tools/VSWhere/VSWhereSettings.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Cake.Core.Tooling;
+
+namespace Cake.Common.Tools.VSWhere
+{
+    /// <summary>
+    /// Contains settings used by <see cref="VSWhereRunner"/>
+    /// </summary>
+    public sealed class VSWhereSettings : ToolSettings
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether to find all instances regardless if they are complete.
+        /// </summary>
+        public bool All { get; set; }
+        
+        /// <summary>
+        /// Gets or sets the products to find. Defaults to Community, Professional, and Enterprise. Specify "*" by itself to search all product instances installed.
+        /// </summary>
+        public List<string> Products { get; set; }
+
+        /// <summary>
+        /// Gets or sets the workload(s) or component(s) required when finding instances.
+        /// </summary>
+        public List<string> Requires { get; set; }
+
+        /// <summary>
+        /// Gets or sets s version range for instances to find. Example: ["15.0","16.0"] will find versions 15.*.
+        /// </summary>
+        public List<string> Version { get; set; }
```

Comment:
Prob just a string? I think it's like nuget's version range (am not 100% sure). Does cake have a type for that if it is?